DalStatement.php

```diff
@@ -47,6 +47,115 @@
 class DalStatement
 {
     /**
+     * Fetch the next row in the result set.
+     *
+     * @var int
+     */
+    const FETCH_FORWARD      = 0;
+
+    /**
+     * Fetch the previous row in the result set.
+     *
+     * @var int
+     */
+    const FETCH_BACKWARD     = 1;
+
+    /**
+     * Start at the first offset.
+     *
+     * @var int
+     */
+    const FETCH_OFFSET_FIRST = 0;
+
+    /**
+     * Start at the last offset.
+     *
+     * @var int
+     */
+    const FETCH_OFFSET_LAST  = -1;
+
+    /**
+     * Specifies that the fetch method shall return each row as an object with
+     * variable names that correspond to the column names returned in the result
+     * set. FETCH_LAZY creates the object variable names as they are accessed.
+     *
+     * @var int
+     */
+    const FETCH_LAZY  = 1;
```

Comment:
So only one object is created and data are updated for each row, is that correct? If yes, lazy object could be correct. What about `AS_REUSABLE_OBEJCT` or something similar (maybe it's a bad idea, I understand “lazy” but not sure everyone will).